src/transformStateWithClones.js

```diff
@@ -66,6 +66,36 @@
  */
 function transformStateWithClones(state, transforms) {
   // write code here
+  const stateClone = { ...state };
+  const states = [];
+
+  for (const transform of transforms) {
+    if (transform.operation === 'addProperties') {
+      for (const property in transform.properties) {
+        stateClone[property] = transform.properties[property];
+      }
+
+      states.push({ ...stateClone });
+    } else if (transform.operation === 'removeProperties') {
+      for (const property of transform.properties) {
+        if (property in stateClone) {
+          delete stateClone[property];
+        }
+      }
+
+      states.push({ ...stateClone });
+    } else if (transform.operation === 'clear') {
+      for (const property in stateClone) {
+        delete stateClone[property];
+      }
+
+      states.push({ ...stateClone });
+    } else {
+      continue;
+    }
```

Comment:
You can write `states.push({ ...stateClone })` only once in the end of each loop iteration.

Switch case construction is more appropriate for this task

Reply:
Yes, with switch case it looks much more better